### PR TITLE
chore: cherry-pick changes from main to release

### PR DESCRIPTION
**Pull request by joshua-goldstein**
Cherry picks:
#1692 
#1790 
#1701 
#1724 
#1702 
#1766 
part of #1808 
#1853 
update changelog

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -476,7 +476,8 @@ func (opt Options) WithBaseTableSize(val int64) Options {
 //
 // LevelSizeMultiplier sets the ratio between the maximum sizes of contiguous levels in the LSM.
 // Once a level grows to be larger than this ratio allowed, the compaction process will be
-//  triggered.
+//
```
Comment (mangalaman93) [inline options.go:479]: remove the extra line

File: options.go [hunk 2]
```diff
@@ -514,7 +515,7 @@ func (opt Options) WithValueThreshold(val int64) Options {
 // and only 1 percent in vlog. The value threshold will be dynamically updated within the range of
 // [ValueThreshold, Options.maxValueThreshold]
 //
-// Say VLogPercentile with 1.0 means threshold will eventually set to Options.maxValueThreshold
+// # Say VLogPercentile with 1.0 means threshold will eventually set to Options.maxValueThreshold
```
Comment (mangalaman93) [inline options.go:518]: why # here?

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

File: options.go (hunk 1 above)
Comment (joshua-goldstein) [reply options.go]: done

File: options.go (hunk 2 above)
Comment (joshua-goldstein) [reply options.go]: not sure how that got in

**Review by mangalaman93 — APPROVED** [review]

**Pull request merged**